src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1648,10 +1648,16 @@ bool esbmc_parseoptionst::process_goto_program(
     namespacet ns(context);
 
     // Start by removing all no-op instructions and unreachable code
-    if(!cmdline.isset("no-remove-no-op"))
+    // We should skip this removal in goto-cov and multi-property
+    if(!(cmdline.isset("no-remove-no-op") || cmdline.isset("multi-property") ||
+         cmdline.isset("parallel-solving") || cmdline.isset("goto-coverage") ||
+         cmdline.isset("goto-coverage-claims")))
       remove_no_op(goto_functions);
 
-    if(!cmdline.isset("no-remove-unreachable"))
+    if(!(cmdline.isset("no-remove-unreachable") ||
+         cmdline.isset("multi-property") || cmdline.isset("parallel-solving") ||
+         cmdline.isset("goto-coverage") ||
+         cmdline.isset("goto-coverage-claims")))
       remove_unreachable(goto_functions);
```

Comment:
Why do any of these newly added options prohibit removing unreachable code?

Reply:
Explained as above.